src/ipfs/tests.rs

```diff
@@ -0,0 +1,122 @@
+use std::thread;
+use std::time::Duration;
+
+use colored::Colorize;
+use futures::executor::block_on;
+use serial_test::serial;
+
+use self::daemon::IpfsDaemon;
+use super::*;
+use crate::ipfs::Ipfs;
+use crate::utils::file_management;
+
+
+fn run_ipfs_test<T>(test: T) -> ()
+    where T: FnOnce(&IpfsDaemon) -> bool
+{
+    let ipfs = IpfsDaemon::default();
+    // for peer in PEER_ADDRS {
+    //     block_on(ipfs.connect_to(peer)).unwrap();
+    //     println!("Connected to peer! {}", peer);
+    // }
+    block_on(ipfs.launch()).unwrap();
+    let has_passed = test(&ipfs);
+    ipfs.shutdown().unwrap();
+    assert!(has_passed)
+}
+
+fn are_files_uploaded(uploaded_paths:Vec<String>, os_paths:Vec<String>) -> bool{
+    let mut is_uploaded = true;
+    for os_path in os_paths{
+        let mut fixed_os_path = String::new();
+        for (i, seg) in os_path.split("/").enumerate() {
+            if seg != "." && !seg.is_empty() {
+                fixed_os_path += &(match i {
+                    1 => String::new(),
+                    2 => seg.to_string(),
+                    _ => "/".to_string() + seg
+                });
+            }
+        }
+
+        let mut is_any_matching = false;
+        'any: for uploaded_path in &uploaded_paths {
+            if &fixed_os_path == uploaded_path {
+                println!("{} == {}", fixed_os_path.green(), uploaded_path.green());
+                is_any_matching = true;
+                break 'any;
+            }else{
```

Comment:
```suggestion
            } else {
```
Could we use spaces after the `}` and before the `}`. This appears to be the style used by tokio, for example: https://github.com/tokio-rs/tokio/blob/c6552c5680fa14105547cfbbc26b26d67197b64e/tokio/src/runtime/context.rs#L207

Reply:
Ran `cargo fmt` fixed mot of that kind of thing